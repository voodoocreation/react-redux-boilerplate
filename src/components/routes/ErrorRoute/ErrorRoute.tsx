import { NextContext } from "next";
import * as React from "react";
import { InjectedIntlProps } from "react-intl";

import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";

import ErrorPage from "../../presentation/ErrorPage/ErrorPage";

interface IResError extends Error {
  statusCode: number;
}

interface IProps extends InjectedIntlProps {
  err: IResError;
  message?: string;
  statusCode: number;
}

class ErrorRoute extends React.Component<IProps> {
  public static getInitialProps(context: NextContext) {
    const res = context.res;
    const err = context.err as IResError;
    const statusCode = res ? res.statusCode : err.statusCode;
    const message = err && err.message ? err.message : undefined;

    return { message, statusCode };
  }

  public render() {
    return (
      <ErrorPage message={this.props.message} status={this.props.statusCode} />
    );
  }
}

const ErrorRouteWrapped = injectIntlIntoPage(ErrorRoute);

export default ErrorRouteWrapped;
