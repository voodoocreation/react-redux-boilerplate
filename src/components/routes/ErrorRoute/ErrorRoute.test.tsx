import { render } from "enzyme";
import * as React from "react";

import ErrorRoute from "./ErrorRoute";

const setup = (fn: any, fromTestProps = {}) => {
  const props = { ...fromTestProps };

  return {
    actual: fn(<ErrorRoute {...props} />),
    props
  };
};

describe("[routes] <ErrorRoute />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);

    expect(actual).toMatchSnapshot();
  });

  it("renders correctly with custom error message", () => {
    const { actual } = setup(render, { message: "Test" });

    expect(actual).toMatchSnapshot();
  });

  describe("getInitialProps()", () => {
    it("gets status code from HTTP response", async () => {
      const props = await ErrorRoute.getInitialProps({
        ctx: {
          res: { statusCode: 404 }
        }
      });

      expect(props.statusCode).toBe(404);
    });

    it("gets status code and error message from context", async () => {
      const props = await ErrorRoute.getInitialProps({
        ctx: {
          err: {
            message: "Error",
            statusCode: 500
          }
        }
      });

      expect(props.message).toBe("Error");
      expect(props.statusCode).toBe(500);
    });

    it("fails silently when status code isn't defined", async () => {
      const props = await ErrorRoute.getInitialProps({
        ctx: {
          err: {
            message: "Error"
          }
        }
      });

      expect(props.message).toBe("Error");
      expect(props.statusCode).toBe(undefined);
    });
  });
});
