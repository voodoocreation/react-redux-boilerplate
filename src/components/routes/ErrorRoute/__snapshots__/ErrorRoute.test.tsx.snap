// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <ErrorRoute /> renders correctly 1`] = `
<article
  class="ErrorPage"
>
  <h1>
    An error has occurred
  </h1>
  <div
    class="ErrorPage--content container"
  >
    <p>
      An error has occurred, please try again.
    </p>
  </div>
</article>
`;
